ops/framework.py

```diff
@@ -489,9 +489,21 @@ def _event_kinds(self) -> list[str]:
                 event_kinds.append(attr_name)
         return event_kinds
 
-    def events(self) -> dict[str, BoundEvent]:
-        """Return a mapping of event_kinds to bound_events for all available events."""
-        return {event_kind: getattr(self, event_kind) for event_kind in self._event_kinds()}
+    def events(self, *event_type: type[EventBase]) -> dict[str, BoundEvent]:
+        """Return a mapping of event names to BoundEvent instances the framework can observe.
+
+        Args:
+            event_type: An optional tuple of event types to narrow the selection.
```

Comment:
This isn't actually passing a tuple, it's *args. @benhoyt indicated he'd rather have a tuple (so keep the doc as-is and change the signature to be `event_type: tuple[type[EventBase]]|None=None` (maybe `event_types`?). It does seem like a tuple rather than *args leaves things more open to having other arguments in the future, even though `events(ops.RelationEvent, ops.WorkloadEvent)` does read nicer than `events((ops.RelationEvent, ops.WorkloadEvent))`.

Reply:
Sorry if I was unclear, but what I was keen on was allowing either a single class or a tuple of classes, just like `issubclass` allows. That makes it the same as `issubclass`, and simple for the single-class case as well as clear for multiple classes; it also allows for additional positional arguments in future (not that I think that's likely).

The type annotation would be `event_type: type[EventBase] | tuple[type[EventBase], ...] | None = None`, and usage would be:

```python
on.events()
on.events(ops.RelationEvent)
on.events((ops.RelationEvent, ops.SecretEvent))
```